src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -388,6 +432,42 @@ private JsonRpcRequest ReadRequest(JToken json)
                 args is JArray argsArray ? (object)PartiallyParsePositionalArguments(argsArray) :
                 null;
 
+            // If method is $/progress, get the progress instance from the dictionary and call Report
+            string method = json.Value<string>("method");
+
+            if (string.Equals(method, ProgressRequestSpecialMethod))
+            {
```

Comment:
tip: It's usually a good idea to explicitly specify the optional 3rd argument: `StringComparer.Ordinal`. I believe this is the default, but especially for protocols it's nice to be explicit so someone else doesn't come in and put `OrdinalIgnoreCase` assuming that that's what you meant.